### ISSUE TITLE
[no-namespace] False positive on namespace inside a declare scope with AllowDeclarations being true

### ISSUE THREAD

**Issue by alex-ubitec** [typescript-eslint/typescript-eslint#2393]
**Repro**
```JSON
{
  "rules": {
   "@typescript-eslint/no-namespace": ["error", { "allowDeclarations": true }]
  }
}
```

```TS
declare global {
  namespace jest {
// ^^^
// ES2015 module syntax is preferred over custom TypeScript modules and namespaces.eslint@typescript-eslint/no-namespa
    interface Matchers<R> {
      toHaveResponseStatus: (min: number, _max?: number) => R;
    }
  }
}

```

**Expected Result** It doesn't throw an error, since we are inside a declare scope

**Actual Result** It reports it as an error

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.34.0` |
| `@typescript-eslint/parser`        | `2.34.0` |
| `TypeScript`                       | `3.9.7` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `13.14.0` |
| `npm`                              | `6.14.4` |

**Comment by alex-ubitec** [comment]
I believe this is maybe identical to  #2238, if it is, I am not sure how to ask to reopen the other issue.

**Comment by bradzacher** [comment]
Please upgrade to a newer version of our tooling and try again.

**Comment by alex-ubitec** [comment]
After upgrading the bug was fixed, sorry for the inconvenience, I didn't realize I was in an old version.

**Comment by bradzacher** [comment]
There's a new version every single monday, so please check for updates before filing an issue!
